### PR TITLE
move from jsonrpc-http-server to hyper

### PR DESCRIPTION
**Pull request by azban**
fixes 1f0f6fecf80aaa5bdb141fa80e5f2af2113aec63

this swaps out the jsonrpc http server we had for a simple hyper http server. it is not very robust and the error handling and responses are pretty shit, but it fixes our issues for now. we need to decide if we are going to move forward with a REST http server or want to move forward with a server that meets JSON-RPC spec. if we want to do the latter, i think the best option is to invest the time in building a JSON-RPC transport for [tarpc](https://github.com/google/tarpc) based on hyper. otherwise, we will need to invest in making our current hyper server more robust and remove a lot of boilerplate that i intentionally left in.

i also didn't bother renaming everything from rpc -> http for now, but we should also do that whenever decide on what we are going to be doing.

### CODE REVIEW

**Comment by nearmax** [comment]
There is a chance we can keep the current HTTP server and add JSON-RPC to it. JSON-RPC is transport-independent, see https://www.jsonrpc.org/specification . There is also a JSON-RPC over HTTP standard: https://www.jsonrpc.org/historical/json-rpc-over-http.html Besides, most of the JSON-RPC clients are working over HTTP anyway.

There is a good chance we can still use the current http server, but then decode/encode the data using `codec.rs` and `message.rs` from https://github.com/vorner/tokio-jsonrpc/tree/master/src . We could write our own encoder/decoder by reading the JSON-RPC standard and would probably arrive to exactly the same code. These encoder/decoder are nicely written in my opinion.

File: node/http/src/api.rs [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+                Ok(CallViewFunctionResponse { result })
+            }
+            Err(e) => { Err(e.to_string()) }
+        }
+    }
+
+    pub fn submit_transaction(&self, r: SignedTransaction) -> Result<(), ()> {
+        debug!(target: "near-rpc", "Received transactio\
+        n {:?}", r);
+        self.submit_txn_sender.clone().try_send(r).unwrap();
```
Comment (nearmax) [inline node/http/src/api.rs]: This will fail if the channel is full. Should return some meaningful error message if that happens instead of panicking.

**Comment by azban** [comment]
yep. the idea behind moving to tarpc is just to have better semantics on the rust side for creating a server and also a rust client if we ever want one. there will probably be an intermediate step in which  we add the encoding and decoding you mentioned to this implementation.

**Comment by azban** [comment]
I personally think this context is more suitable for RPC than REST. I'll open up an issue for discussion as it's outside scope of this PR.

**Comment by vgrichina** [comment]
Is this branch supposed to work as drop in replacement or are there any API changes?

**Comment by azban** [comment]
> Is this branch supposed to work as drop in replacement or are there any API changes?

this is a drop in replacement for the API, but the script usage has changed slightly. args are now specified with `--args`  as they are optional.

**Pull request merged**